node/src/request_coordinator.ts

```diff
@@ -0,0 +1,65 @@
+import {promises as DNS, SrvRecord} from 'dns';
+import {AdminClient} from './admin_client';
+
+/**
+ * Supports to send pause/unpause requests to all hosts of a service records
+ */
+export class RequestCoordinator {
```

Comment:
This class supports to send pause/unpause requests to the hosts behind a service record.
If it fails to pause any of the hosts, it will tries to unpause (as a rollback process) all of them.